libs/antsi/src/font.rs

```diff
@@ -106,25 +109,57 @@ pub enum Blinking {
 }
 
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[cfg(feature = "script")]
 #[non_exhaustive]
 pub enum FontScript {
     Sub,
     Super,
 }
 
+macro_rules! impl_builder {
+    (
+        $(#[$meta_set:meta])*
+        $set:ident($set_var:ident : $set_ty:ty);
+        $(#[$meta_with:meta])*
+        $with:ident($with_var:ident : $with_ty:ty);
+        $(#[$meta_getter:meta])*
+        $get:ident() -> $ret:ty
+    ) => {
+        $(#[$meta_set])*
+        pub fn $set(&mut self, $set_var: $set_ty) -> &mut Self {
+            self.$set_var = Some($set_var);
+            self
+        }
+
+        $(#[$meta_with])*
+        #[must_use]
+        pub const fn $with(mut self, $with_var: $with_ty) -> Self {
+            self.$with_var = Some($with_var);
```

Comment:
As proposed above, this should be implemented without a macro, but if we would keep the setters, can't you re-use those?
```rust
self.$set($with_ty);
self
```